2025-03-26 16:59:40.837840: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcudart.so.10.1'; dlerror: libcudart.so.10.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/easybuild/software/Python/3.8.6-GCCcore-12.2.0/lib:/opt/easybuild/software/OpenSSL/1.1/lib:/opt/easybuild/software/libffi/3.4.4-GCCcore-12.2.0/lib64:/opt/easybuild/software/GMP/6.2.1-GCCcore-12.2.0/lib:/opt/easybuild/software/XZ/5.2.7-GCCcore-12.2.0/lib:/opt/easybuild/software/SQLite/3.39.4-GCCcore-12.2.0/lib:/opt/easybuild/software/Tcl/8.6.12-GCCcore-12.2.0/lib:/opt/easybuild/software/libreadline/8.2-GCCcore-12.2.0/lib:/opt/easybuild/software/ncurses/6.3-GCCcore-12.2.0/lib:/opt/easybuild/software/bzip2/1.0.8-GCCcore-12.2.0/lib:/opt/easybuild/software/binutils/2.39-GCCcore-12.2.0/lib:/opt/easybuild/software/zlib/1.2.12-GCCcore-12.2.0/lib:/opt/easybuild/software/GCCcore/12.2.0/lib64
2025-03-26 16:59:40.837868: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2025-03-26 16:59:43.299402: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcuda.so.1
2025-03-26 16:59:43.348190: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2025-03-26 16:59:43.348484: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: 
pciBusID: 0000:2c:00.0 name: NVIDIA GeForce RTX 3090 computeCapability: 8.6
coreClock: 1.695GHz coreCount: 82 deviceMemorySize: 23.69GiB deviceMemoryBandwidth: 871.81GiB/s
2025-03-26 16:59:43.348555: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2025-03-26 16:59:43.348793: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 1 with properties: 
pciBusID: 0000:61:00.0 name: NVIDIA GeForce RTX 3090 computeCapability: 8.6
coreClock: 1.695GHz coreCount: 82 deviceMemorySize: 23.69GiB deviceMemoryBandwidth: 871.81GiB/s
2025-03-26 16:59:43.348928: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcudart.so.10.1'; dlerror: libcudart.so.10.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/easybuild/software/Python/3.8.6-GCCcore-12.2.0/lib:/opt/easybuild/software/OpenSSL/1.1/lib:/opt/easybuild/software/libffi/3.4.4-GCCcore-12.2.0/lib64:/opt/easybuild/software/GMP/6.2.1-GCCcore-12.2.0/lib:/opt/easybuild/software/XZ/5.2.7-GCCcore-12.2.0/lib:/opt/easybuild/software/SQLite/3.39.4-GCCcore-12.2.0/lib:/opt/easybuild/software/Tcl/8.6.12-GCCcore-12.2.0/lib:/opt/easybuild/software/libreadline/8.2-GCCcore-12.2.0/lib:/opt/easybuild/software/ncurses/6.3-GCCcore-12.2.0/lib:/opt/easybuild/software/bzip2/1.0.8-GCCcore-12.2.0/lib:/opt/easybuild/software/binutils/2.39-GCCcore-12.2.0/lib:/opt/easybuild/software/zlib/1.2.12-GCCcore-12.2.0/lib:/opt/easybuild/software/GCCcore/12.2.0/lib64
2025-03-26 16:59:43.349008: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcublas.so.10'; dlerror: libcublas.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/easybuild/software/Python/3.8.6-GCCcore-12.2.0/lib:/opt/easybuild/software/OpenSSL/1.1/lib:/opt/easybuild/software/libffi/3.4.4-GCCcore-12.2.0/lib64:/opt/easybuild/software/GMP/6.2.1-GCCcore-12.2.0/lib:/opt/easybuild/software/XZ/5.2.7-GCCcore-12.2.0/lib:/opt/easybuild/software/SQLite/3.39.4-GCCcore-12.2.0/lib:/opt/easybuild/software/Tcl/8.6.12-GCCcore-12.2.0/lib:/opt/easybuild/software/libreadline/8.2-GCCcore-12.2.0/lib:/opt/easybuild/software/ncurses/6.3-GCCcore-12.2.0/lib:/opt/easybuild/software/bzip2/1.0.8-GCCcore-12.2.0/lib:/opt/easybuild/software/binutils/2.39-GCCcore-12.2.0/lib:/opt/easybuild/software/zlib/1.2.12-GCCcore-12.2.0/lib:/opt/easybuild/software/GCCcore/12.2.0/lib64
2025-03-26 16:59:43.349090: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcufft.so.10'; dlerror: libcufft.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/easybuild/software/Python/3.8.6-GCCcore-12.2.0/lib:/opt/easybuild/software/OpenSSL/1.1/lib:/opt/easybuild/software/libffi/3.4.4-GCCcore-12.2.0/lib64:/opt/easybuild/software/GMP/6.2.1-GCCcore-12.2.0/lib:/opt/easybuild/software/XZ/5.2.7-GCCcore-12.2.0/lib:/opt/easybuild/software/SQLite/3.39.4-GCCcore-12.2.0/lib:/opt/easybuild/software/Tcl/8.6.12-GCCcore-12.2.0/lib:/opt/easybuild/software/libreadline/8.2-GCCcore-12.2.0/lib:/opt/easybuild/software/ncurses/6.3-GCCcore-12.2.0/lib:/opt/easybuild/software/bzip2/1.0.8-GCCcore-12.2.0/lib:/opt/easybuild/software/binutils/2.39-GCCcore-12.2.0/lib:/opt/easybuild/software/zlib/1.2.12-GCCcore-12.2.0/lib:/opt/easybuild/software/GCCcore/12.2.0/lib64
2025-03-26 16:59:43.349165: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcurand.so.10'; dlerror: libcurand.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/easybuild/software/Python/3.8.6-GCCcore-12.2.0/lib:/opt/easybuild/software/OpenSSL/1.1/lib:/opt/easybuild/software/libffi/3.4.4-GCCcore-12.2.0/lib64:/opt/easybuild/software/GMP/6.2.1-GCCcore-12.2.0/lib:/opt/easybuild/software/XZ/5.2.7-GCCcore-12.2.0/lib:/opt/easybuild/software/SQLite/3.39.4-GCCcore-12.2.0/lib:/opt/easybuild/software/Tcl/8.6.12-GCCcore-12.2.0/lib:/opt/easybuild/software/libreadline/8.2-GCCcore-12.2.0/lib:/opt/easybuild/software/ncurses/6.3-GCCcore-12.2.0/lib:/opt/easybuild/software/bzip2/1.0.8-GCCcore-12.2.0/lib:/opt/easybuild/software/binutils/2.39-GCCcore-12.2.0/lib:/opt/easybuild/software/zlib/1.2.12-GCCcore-12.2.0/lib:/opt/easybuild/software/GCCcore/12.2.0/lib64
2025-03-26 16:59:43.349238: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcusolver.so.10'; dlerror: libcusolver.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/easybuild/software/Python/3.8.6-GCCcore-12.2.0/lib:/opt/easybuild/software/OpenSSL/1.1/lib:/opt/easybuild/software/libffi/3.4.4-GCCcore-12.2.0/lib64:/opt/easybuild/software/GMP/6.2.1-GCCcore-12.2.0/lib:/opt/easybuild/software/XZ/5.2.7-GCCcore-12.2.0/lib:/opt/easybuild/software/SQLite/3.39.4-GCCcore-12.2.0/lib:/opt/easybuild/software/Tcl/8.6.12-GCCcore-12.2.0/lib:/opt/easybuild/software/libreadline/8.2-GCCcore-12.2.0/lib:/opt/easybuild/software/ncurses/6.3-GCCcore-12.2.0/lib:/opt/easybuild/software/bzip2/1.0.8-GCCcore-12.2.0/lib:/opt/easybuild/software/binutils/2.39-GCCcore-12.2.0/lib:/opt/easybuild/software/zlib/1.2.12-GCCcore-12.2.0/lib:/opt/easybuild/software/GCCcore/12.2.0/lib64
2025-03-26 16:59:43.349309: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcusparse.so.10'; dlerror: libcusparse.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/easybuild/software/Python/3.8.6-GCCcore-12.2.0/lib:/opt/easybuild/software/OpenSSL/1.1/lib:/opt/easybuild/software/libffi/3.4.4-GCCcore-12.2.0/lib64:/opt/easybuild/software/GMP/6.2.1-GCCcore-12.2.0/lib:/opt/easybuild/software/XZ/5.2.7-GCCcore-12.2.0/lib:/opt/easybuild/software/SQLite/3.39.4-GCCcore-12.2.0/lib:/opt/easybuild/software/Tcl/8.6.12-GCCcore-12.2.0/lib:/opt/easybuild/software/libreadline/8.2-GCCcore-12.2.0/lib:/opt/easybuild/software/ncurses/6.3-GCCcore-12.2.0/lib:/opt/easybuild/software/bzip2/1.0.8-GCCcore-12.2.0/lib:/opt/easybuild/software/binutils/2.39-GCCcore-12.2.0/lib:/opt/easybuild/software/zlib/1.2.12-GCCcore-12.2.0/lib:/opt/easybuild/software/GCCcore/12.2.0/lib64
2025-03-26 16:59:43.349384: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcudnn.so.7'; dlerror: libcudnn.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/easybuild/software/Python/3.8.6-GCCcore-12.2.0/lib:/opt/easybuild/software/OpenSSL/1.1/lib:/opt/easybuild/software/libffi/3.4.4-GCCcore-12.2.0/lib64:/opt/easybuild/software/GMP/6.2.1-GCCcore-12.2.0/lib:/opt/easybuild/software/XZ/5.2.7-GCCcore-12.2.0/lib:/opt/easybuild/software/SQLite/3.39.4-GCCcore-12.2.0/lib:/opt/easybuild/software/Tcl/8.6.12-GCCcore-12.2.0/lib:/opt/easybuild/software/libreadline/8.2-GCCcore-12.2.0/lib:/opt/easybuild/software/ncurses/6.3-GCCcore-12.2.0/lib:/opt/easybuild/software/bzip2/1.0.8-GCCcore-12.2.0/lib:/opt/easybuild/software/binutils/2.39-GCCcore-12.2.0/lib:/opt/easybuild/software/zlib/1.2.12-GCCcore-12.2.0/lib:/opt/easybuild/software/GCCcore/12.2.0/lib64
2025-03-26 16:59:43.349396: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1753] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2025-03-26 16:59:43.349695: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2025-03-26 16:59:43.375896: I tensorflow/core/platform/profile_utils/cpu_utils.cc:104] CPU Frequency: 3493535000 Hz
2025-03-26 16:59:43.376997: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x226e670 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2025-03-26 16:59:43.377010: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2025-03-26 16:59:43.377970: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1257] Device interconnect StreamExecutor with strength 1 edge matrix:
2025-03-26 16:59:43.377976: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1263]      
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.
Traceback (most recent call last):
  File "Packet_based_RePO_Adversarial.py", line 282, in <module>
    temp = find_adv(np.copy(x))
  File "Packet_based_RePO_Adversarial.py", line 148, in find_adv
    res = delay_and_split(np.copy(x))
  File "Packet_based_RePO_Adversarial.py", line 161, in delay_and_split
    adv_x = get_delayed_splited(x,len_last)
  File "/home/CIN/lims/RePO/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py", line 780, in __call__
    result = self._call(*args, **kwds)
  File "/home/CIN/lims/RePO/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py", line 814, in _call
    results = self._stateful_fn(*args, **kwds)
  File "/home/CIN/lims/RePO/lib/python3.8/site-packages/tensorflow/python/eager/function.py", line 2829, in __call__
    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
  File "/home/CIN/lims/RePO/lib/python3.8/site-packages/tensorflow/python/eager/function.py", line 1843, in _filtered_call
    return self._call_flat(
  File "/home/CIN/lims/RePO/lib/python3.8/site-packages/tensorflow/python/eager/function.py", line 1923, in _call_flat
    return self._build_call_outputs(self._inference_function.call(
  File "/home/CIN/lims/RePO/lib/python3.8/site-packages/tensorflow/python/eager/function.py", line 545, in call
    outputs = execute.execute(
  File "/home/CIN/lims/RePO/lib/python3.8/site-packages/tensorflow/python/eager/execute.py", line 59, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.FailedPreconditionError:  Error while reading resource variable _AnonymousVar7 from Container: localhost. This could mean that the variable was uninitialized. Not found: Resource localhost/_AnonymousVar7/N10tensorflow3VarE does not exist.
	 [[node Maximum_1/ReadVariableOp (defined at Packet_based_RePO_Adversarial.py:68) ]] [Op:__inference_get_delayed_splited_8632]

Function call stack:
get_delayed_splited

